services/QuillLMS/spec/services/classlink_integration/classroom_persister_spec.rb

```diff
@@ -0,0 +1,83 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe ClasslinkIntegration::ClassroomPersister do
+  let(:incoming_classroom) do
+    {
+      sourced_id: 'external-123',
+      name: 'Math 101',
+      visible: true
+    }
+  end
+
+  describe '#run' do
+    let(:classlink_instance) { create(:classlink_instance) }
+
+    context 'when the classroom already exists' do
+      let(:existing_classroom) { create(:classroom, name: 'Old Math Class', visible: false) }
+      let(:classlink_classroom) do
+        create(:classlink_classroom,
+          external_id: 'external-123',
+          classroom: existing_classroom,
+          provider_instance: classlink_instance,
+          type: ClasslinkClassroom)
+      end
+      let(:existing_provider_classrooms) { [classlink_classroom] }
+
+      it 'updates the attributes of the existing classroom' do
+        persister = described_class.new(incoming_classroom:, existing_provider_classrooms:, classlink_instance_id: classlink_instance.id)
+        expect { persister.run }.to not_change(ClasslinkClassroom, :count)
+
+        existing_classroom.reload
+        expect(existing_classroom.name).to eq('Math 101')
+        expect(existing_classroom.visible).to be true
+      end
+    end
+
+    context 'classroom does not exist' do
+      existing_provider_classrooms = []
```

Comment:
Using a `let` block for variable assignment outside of test block is more conventional for RSpec.